File: app/src/main/java/privacyfriendlyexample/org/secuso/boardgameclock/model/Player.java
package privacyfriendlyexample.org.secuso.boardgameclock.model;

import android.graphics.Bitmap;

import java.text.SimpleDateFormat;
import java.util.Date;

public class Player {

    private long id;
    private long date;
    private String dateString;
    private String name;
    private Bitmap icon;

    public Player(long id, long date, String name, Bitmap icon) {
        this.id = id;
        this.date = date;
        this.name = name;
        this.icon = icon;
    }

    public Player() {

    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Bitmap getIcon() {
        return icon;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public void setIcon(Bitmap icon) {
        this.icon = icon;
    }

    @Override
    public String toString(){
        return name;
    }

    public void setDate(long date) {
        this.date = date;

        SimpleDateFormat formatter = new SimpleDateFormat("dd.MM.yyyy kk:mm");
        dateString = formatter.format(new Date(date));
    }

    public String getDateString(){
        return dateString;
    }

}

File: app/src/main/java/org/secuso/privacyfriendlyboardgameclock/room/model/Player.kt
/*
 This file is part of Privacy Friendly Board Game Clock.

 Privacy Friendly Board Game Clock is free software:
 you can redistribute it and/or modify it under the terms of the
 GNU General Public License as published by the Free Software Foundation,
 either version 3 of the License, or any later version.

 Privacy Friendly App Example is distributed in the hope
 that it will be useful, but WITHOUT ANY WARRANTY; without even
 the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 See the GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with Privacy Friendly App Example. If not, see <http://www.gnu.org/licenses/>.
 */
package org.secuso.privacyfriendlyboardgameclock.room.model

import android.graphics.Bitmap
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import java.text.SimpleDateFormat
import java.util.Date

/**
 * Created by Quang Anh Dang on 24.12.2017.
 * @author Quang Anh Dang
 * Last changed on 18.03.18
 * Model for Player Object
 */
@Entity(tableName = "players")
data class Player(
    @PrimaryKey
    @ColumnInfo(name = "_id") var id: Long = 0,
    @ColumnInfo(name = "name") var name: String = "",
    @ColumnInfo(name = "icon") var icon: Bitmap? = null

) {
    @ColumnInfo(name = "date", defaultValue = "CURRENT_TIMESTAMP") var date: Long = 0
        set(date) {
            field = date
            dateString = SimpleDateFormat("dd.MM.yyyy kk:mm").format(Date(date))
        }
    @Ignore var dateString: String = ""
        private set
}
